src/main/java/org/elasticsearch/repositories/RepositoriesService.java

```diff
@@ -190,6 +190,9 @@ public ClusterState execute(ClusterState currentState) {
                         mdBuilder.putCustom(RepositoriesMetaData.TYPE, repositories);
                         return ClusterState.builder(currentState).metaData(mdBuilder).build();
                     }
+                    if (Regex.isMatchAllPattern(request.name)) { // we use a wildcard so we don't barf if it's not present.
```

Comment:
This code will be never executed. I think you meant to put it after the if statement where we check if we have any repositories registered.